### PR TITLE
Small fixes for slice edge cases

### PR DESCRIPTION
**Pull request by dan-garvey**
Adds a check for optional type, and covers the case where end < start. Thanks to @cathyzhyi for pointing these out.

### CODE REVIEW

**Comment by silvasean** [comment]
Please add tests showing the new functionality :)

**Comment by dan-garvey** [comment]
Unfortunately this case goes from xfail to xfail, as it results in 0 size tensors which still crash the refbackend #448

Do we want to create tests for xfail on Optional type coverage?

**Comment by silvasean** [comment]
is there a way to hack around that temporarily? for example, to cat the zero size tensor with a tensor of size 1 so the final tensor is of size 1?

**Comment by dan-garvey** [comment]
Have they tried cloning you yet

File: e2e_testing/torchscript/slice_like.py [hunk 1]
```diff
@@ -121,7 +126,10 @@ def __init__(self):
         ([-1, -1, -1], torch.float32, True),
     ])
     def forward(self, x):
-        return x[5:5, 3:3, -1:]
+        #TODO: remove hacky cat tensor once refbackend supports 0 size dim
```
Comment (silvasean) [inline e2e_testing/torchscript/slice_like.py:129]: nit: space before TODO

**Review by silvasean — APPROVED** [review]
Go ahead and merge after addressing nit :)

**Pull request merged**